controller-server/src/main/java/com/yahoo/vespa/hosted/controller/archive/CuratorArchiveBucketDb.java

```diff
@@ -41,23 +44,30 @@
 
     private final ArchiveService archiveService;
     private final CuratorDb curatorDb;
-    private final boolean enabled;
+    private final FlagSource flagSource;
+    private final SystemName system;
 
     public CuratorArchiveBucketDb(Controller controller) {
         this.archiveService = controller.serviceRegistry().archiveService();
         this.curatorDb = controller.curator();
-        SystemName system = controller.zoneRegistry().system();
-        this.enabled = system.isPublic() || system.isCd();
+        this.flagSource = controller.flagSource();
+        this.system = controller.zoneRegistry().system();
     }
 
     public Optional<URI> archiveUriFor(ZoneId zoneId, TenantName tenant) {
-        if (enabled) {
+        if (enabled(zoneId)) {
             return Optional.of(URI.create(Text.format("s3://%s/%s/", findOrAssignBucket(zoneId, tenant), tenant.value())));
         } else {
             return Optional.empty();
         }
     }
 
+    private boolean enabled(ZoneId zone) {
+        return system.isPublic() || Flags.ENABLE_ONPREM_TENANT_S3_ARCHIVE.bindTo(flagSource)
```

Comment:
This will disable it for CD?

Reply:
It's already effectively disabled as the archive maintainer is disabled through feature flag in cd.